File: Fortran/0156/0156_0002.f90
subroutine s1
type ty
integer(kind=8) :: xx
end type ty
 
type,extends(ty) :: ty1
integer(kind=8) :: yy
end type ty1

class(ty1),pointer :: cptr1(:)
call sub(NULL(cptr1))        
contains

subroutine sub(d_cptr)
class(ty1),pointer ::d_cptr(:)


print*, d_cptr(1)%xx 
end subroutine
end
print *,'pass'
end

File: C/0156/0156_0002.c
#include <stdio.h>
struct x 
{
  int *p;
  int *q;
};
#define N 100

int sub(struct x *a, struct x *b) 
{
  int i;
  for (i=0; i<N; ++i) {
    a->p[i] = i;
    b->q[i] = N-i;
    if (a->p[i] == i) {
      a->p[i] = 0;
    }
  }
  return a->p[10];
}

int main(void) 
{
  struct x u;
  int a[N];
  u.p = a;
  u.q = a;
  if (sub(&u, &u) == 90) {
    printf("ok\n");
  } else {
    printf("ng\n");
  }
}
